src/Azure/Orleans.Transactions.AzureStorage/Hosting/AzureTableTransactionsSiloBuilderExtensions.cs

```diff
@@ -0,0 +1,56 @@
+using Microsoft.Extensions.DependencyInjection;
+using Orleans.Runtime;
+using Orleans.Configuration;
+using Orleans.Transactions.Abstractions;
+using Orleans.Transactions.AzureStorage.TransactionalState;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Orleans.Providers;
+
+namespace Orleans.Hosting
+{
+    public static class AzureTableSiloBuilderExtensions
+    {
+        /// <summary>
+        /// Configure silo to use azure table storage as the default transactional grain storage.
+        /// </summary>
+        public static ISiloHostBuilder AddAzureTableTransactionalStateStorageAsDefault(this ISiloHostBuilder builder, Action<AzureTableTransactionalStateOptions> configureOptions)
+        {
+            return builder.AddAzureTableTransactionalStateStorage(ProviderConstants.DEFAULT_STORAGE_PROVIDER_NAME, configureOptions);
+        }
+
+        /// <summary>
+        /// Configure silo to use azure table storage for transactional grain storage.
+        /// </summary>
+        public static ISiloHostBuilder AddAzureTableTransactionalStateStorage(this ISiloHostBuilder builder, string name, Action<AzureTableTransactionalStateOptions> configureOptions)
+        {
+            return builder.ConfigureServices(services => services.AddAzureTableTransactionalStateStorage(name, ob => ob.Configure(configureOptions)));
+        }
+
+        /// <summary>
+        /// Configure silo to use azure table storage as the default transactional grain storage.
+        /// </summary>
+        public static ISiloHostBuilder AddAzureTableTransactionalStateStorageAsDefault(this ISiloHostBuilder builder, Action<OptionsBuilder<AzureTableTransactionalStateOptions>> configureOptions = null)
+        {
+            return builder.AddAzureTableTransactionalStateStorage(ProviderConstants.DEFAULT_STORAGE_PROVIDER_NAME, configureOptions);
+        }
+
+        /// <summary>
+        /// Configure silo to use azure table storage for transactional grain storage.
+        /// </summary>
+        public static ISiloHostBuilder AddAzureTableTransactionalStateStorage(this ISiloHostBuilder builder, string name, Action<OptionsBuilder<AzureTableTransactionalStateOptions>> configureOptions = null)
+        {
+            return builder.ConfigureServices(services => services.AddAzureTableTransactionalStateStorage(name, configureOptions));
+        }
+
+        private static IServiceCollection AddAzureTableTransactionalStateStorage(this IServiceCollection services, string name,
+            Action<OptionsBuilder<AzureTableTransactionalStateOptions>> configureOptions = null)
+        {
+            configureOptions?.Invoke(services.AddOptions<AzureTableTransactionalStateOptions>(name));
+            services.ConfigureNamedOptionForLogging<AzureTableTransactionalStateOptions>(name);
+            services.TryAddSingleton<ITransactionalStateStorageFactory>(sp => sp.GetServiceByName<ITransactionalStateStorageFactory>(ProviderConstants.DEFAULT_STORAGE_PROVIDER_NAME));
```

Comment:
Should this be `name` here or am I misunderstanding?

Reply:
This is just setting up a mapping for the default, so that if anyone resolves ITransactionalStateStorageFactory it will get the one named ProviderConstants.DEFAULT_STORAGE_PROVIDER_NAME

Reply:
It seems odd, but ok. Maybe it would be better to add a special case for default instead of unconditionally adding a registration to a service which may not exist.

Reply:
This is the pattern used to support legacy providers which indicated a provider was the default by naming it 'default', whether there was a provider named that or not.  I am not at all opposed to refining or altering this pattern, but we should be consistent.  I'll open an issue to review this pattern.